tests/mypy_test.py

```diff
@@ -213,57 +233,46 @@ def run_mypy(args: TestConfig, configurations: list[MypyDistConf], files: list[P
                 temp.write(f"{k} = {v}\n")
         temp.flush()
 
-        flags = get_mypy_flags(args, temp.name, testing_stdlib=testing_stdlib)
+        flags = [
+            "--python-version",
+            args.version,
+            "--show-traceback",
+            "--warn-incomplete-stub",
+            "--show-error-codes",
+            "--no-error-summary",
+            "--platform",
+            args.platform,
+            "--custom-typeshed-dir",
+            str(Path(__file__).parent.parent),
+            "--strict",
+            # Stub completion is checked by pyright (--allow-*-defs)
+            "--allow-untyped-defs",
+            "--allow-incomplete-defs",
+            "--allow-subclassing-any",  # TODO: Do we still need this now that non-types dependencies are allowed?
```

Comment:
For more context: We can't remove this flag until all stubs are updated to use their non-types dependencies. And it will depend if all subclassed dependencies are typed. (although some workaround or a `type: ignore` or two may also work)

Reply:
Maybe we can remove the flag now and `# type: ignore` the errors it generates, then remove them as we start using non-types deps.

Reply:
Here's the results of running mypy on just our stubs for `ldap3` if we remove this flag:

```
testing ldap3 (110 files)... failure

stubs\ldap3\ldap3\utils\asn1.pyi:10: error: Class cannot subclass "_AbstractItemEncoder" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:28: error: Class cannot subclass "Integer" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:31: error: Class cannot subclass "OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:35: error: Class cannot subclass "OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:40: error: Class cannot subclass "OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:43: error: Class cannot subclass "OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:46: error: Class cannot subclass "Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:51: error: Class cannot subclass "SetOf" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:54: error: Class cannot subclass "SetOf" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:58: error: Class cannot subclass "Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:61: error: Class cannot subclass "Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:64: error: Class cannot subclass "SequenceOf" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:67: error: Class cannot subclass "OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:71: error: Class cannot subclass "OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:74: error: Class cannot subclass "Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:78: error: Class cannot subclass "OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:82: error: Class cannot subclass "OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:86: error: Class cannot subclass "OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:90: error: Class cannot subclass "Choice" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:93: error: Class cannot subclass "Integer" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:96: error: Class cannot subclass "Enumerated" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:102: error: Class cannot subclass "SequenceOf" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:106: error: Class cannot subclass "OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:110: error: Class cannot subclass "Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:113: error: Class cannot subclass "Boolean" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:116: error: Class cannot subclass "OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:119: error: Class cannot subclass "Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:122: error: Class cannot subclass "SequenceOf" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:126: error: Class cannot subclass "Enumerated" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:129: error: Class cannot subclass "Enumerated" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:132: error: Class cannot subclass "Boolean" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:135: error: Class cannot subclass "SequenceOf" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:147: error: Class cannot subclass "Boolean" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:151: error: Class cannot subclass "Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:163: error: Class cannot subclass "Choice" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:166: error: Class cannot subclass "SequenceOf" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:170: error: Class cannot subclass "Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:174: error: Class cannot subclass "SetOf" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:178: error: Class cannot subclass "SetOf" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:182: error: Class cannot subclass "Choice" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:202: error: Class cannot subclass "Choice" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:205: error: Class cannot subclass "SequenceOf" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:208: error: Class cannot subclass "Enumerated" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:211: error: Class cannot subclass "Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:214: error: Class cannot subclass "SequenceOf" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:217: error: Class cannot subclass "Boolean" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:225: error: Class cannot subclass "OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:232: error: Class cannot subclass "OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:239: error: Class cannot subclass "OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:243: error: Class cannot subclass "Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:247: error: Class cannot subclass "Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:251: error: Class cannot subclass "Null" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:254: error: Class cannot subclass "Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:258: error: Class cannot subclass "SequenceOf" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:263: error: Class cannot subclass "Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:270: error: Class cannot subclass "Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:277: error: Class cannot subclass "Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:290: error: Class cannot subclass "Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:297: error: Class cannot subclass "Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:307: error: Class cannot subclass "Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:311: error: Class cannot subclass "Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:315: error: Class cannot subclass "Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:319: error: Class cannot subclass "Choice" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc4511.pyi:322: error: Class cannot subclass "Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc3062.pyi:9: error: Class cannot subclass "_OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc3062.pyi:13: error: Class cannot subclass "_OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc3062.pyi:17: error: Class cannot subclass "_OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc3062.pyi:21: error: Class cannot subclass "_OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc3062.pyi:25: error: Class cannot subclass "_Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc3062.pyi:28: error: Class cannot subclass "_Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc2696.pyi:13: error: Class cannot subclass "_Integer" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc2696.pyi:17: error: Class cannot subclass "_OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\rfc2696.pyi:19: error: Class cannot subclass "_Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\persistentSearch.pyi:9: error: Class cannot subclass "_Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\persistentSearch.pyi:12: error: Class cannot subclass "_Enumerated" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\persistentSearch.pyi:15: error: Class cannot subclass "_Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\novell.pyi:13: error: Class cannot subclass "_OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\novell.pyi:16: error: Class cannot subclass "_OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\novell.pyi:20: error: Class cannot subclass "_OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\novell.pyi:24: error: Class cannot subclass "_OctetString" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\novell.pyi:28: error: Class cannot subclass "_Integer" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\novell.pyi:31: error: Class cannot subclass "_Integer" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\novell.pyi:34: error: Class cannot subclass "_Integer" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\novell.pyi:37: error: Class cannot subclass "_Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\novell.pyi:40: error: Class cannot subclass "_Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\novell.pyi:43: error: Class cannot subclass "_Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\novell.pyi:46: error: Class cannot subclass "_Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\novell.pyi:49: error: Class cannot subclass "_SequenceOf" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\novell.pyi:52: error: Class cannot subclass "_Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\novell.pyi:56: error: Class cannot subclass "_Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\novell.pyi:60: error: Class cannot subclass "_Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\novell.pyi:63: error: Class cannot subclass "_Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\novell.pyi:66: error: Class cannot subclass "_Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\novell.pyi:69: error: Class cannot subclass "_Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\novell.pyi:72: error: Class cannot subclass "_Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\microsoft.pyi:8: error: Class cannot subclass "_Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\microsoft.pyi:12: error: Class cannot subclass "_Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\microsoft.pyi:15: error: Class cannot subclass "_Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\microsoft.pyi:18: error: Class cannot subclass "_Sequence" (has type "Any")  [misc]
stubs\ldap3\ldap3\protocol\microsoft.pyi:21: error: Class cannot subclass "_Sequence" (has type "Any")  [misc]
```

Reply:
```suggestion
            "--allow-subclassing-any",  # TODO: Do we still need this now that non-types dependencies are allowed? (#5768)
```

Reply:
Yeah removing `--allow-subclassing-any` isn't going to be done in a single PR. Will probably need a tracking issue (would be an "Epic" in other tracking systems). Exactly like JelleZijlstra did for bytes type.

A lot are just about using non-type dependencies.
Some will require new stubs to be referenced.
And some would be too complex (like Qt types in pillow).

An "exceptions/exclude" approach like in `pyrightconfig.stricter.json` to progressively remove Any subclassing might also be beneficial (and wouldn't require an issue to track).

Reply:
### Only affects themselves
beautifulsoup4 # Fixed by 9462
ldap3 # Fixable after #9437
pika # Will be fixable by importing twisted
pyOpenSSL # fixed by #9459
python-jose # Fixable after #9437
tqdm # Will be partially fixable by importing rich

### Affected by SQLAlchemy, Will be mostly fixable by importing mypy
(greenlet is not typed and will require stubs https://github.com/python-greenlet/greenlet/issues/244)
SQLAlchemy
Flask-Migrate
Flask-SQLAlchemy

### Affected by setuptools, Fixed by #9460
setuptools
pyinstaller # Will be fixable by importing importlib_metadata
Pygments

### Affected by paramiko, fixed by #9459
paramiko
pysftp

### Affected by Pillow, fixed in #9461
Pillow
D3DShot
python-xlib
fpdf2 # https://github.com/fonttools/fonttools/issues/2943
PyScreeze
PyAutoGUI

---

This also means you can look into considering adding the fllowing tot eh stubsuploader allowlist: rich, twisted, importlib_metadata, and mypy